### PR TITLE
[alembic] Enable release only and debug only build on alembic port

### LINKED ISSUE #18962: [alembic] Installing in debug only or in release only is not enable on this version of the port
****

**Is your feature request related to a problem? Please describe.**
The issue concerns this file (https://github.com/microsoft/vcpkg/blob/master/ports/alembic/portfile.cmake)

Whenever I try to install alembic using vcpkg while using a homemade "release only" triplet, I noticed that there was no test to check the VCPKG_BUILD_TYPE in this port. Therefore the build fails because it tries to use debug functions.

**Proposed solution**
Just as it was done in other ports I want to add those VCPKG_BUILD_TYPE test.
I have actually started a PR about this and I'm making this issue to explain what it will fix.
The PR : #18963 

**Describe alternatives you've considered**
 The changes work fine on my end.

**Additional context**
This is my first issue/PR on vcpkg so I apologize if I did something wrong.

Issue

### PR DESCRIPTION
**Pull request by DanielDelaporus**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18962 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
This change doesn't tamper with any of the triplets from the baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes. (The version is now changed)

### CODE REVIEW

**Comment by cenit** [comment]
something bad happened here with port version, registry and tests in ci (everything is green but a big red flag should be there because pr  is so wrong on this side!)

@ras0219-msft @strega-nil

**Comment by dg0yt** [comment]
Port version 1 now refers to the build rules (git-tree) of the former port version 0. I guess that's why CI didn't build anything for port version 1. And port version 0 has new build rules.

**Comment by cenit** [comment]
yes, it just means that tests are not good and that's the reason for the tags [/ot]

**Comment by BillyONeal** [comment]
> yes, it just means that tests are not good and that's the reason for the tags [/ot]

The testing checks that the version database contains the current SHA, it doesn't check that each SHA exists, if that makes sense.

File: ports/alembic/portfile.cmake [hunk 1]
```diff
@@ -45,23 +45,31 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
...
-    string(REPLACE "\${_IMPORT_PREFIX}/debug/lib/Alembic.dll"
-                   "\${_IMPORT_PREFIX}/debug/bin/Alembic.dll" DEBUG_CONFIG "${DEBUG_CONFIG}")
-    file(WRITE ${CURRENT_PACKAGES_DIR}/share/${PORT}/AlembicTargets-debug.cmake "${DEBUG_CONFIG}")
-
-    file(READ ${CURRENT_PACKAGES_DIR}/share/${PORT}/AlembicTargets-release.cmake RELEASE_CONFIG)
-    string(REPLACE "\${_IMPORT_PREFIX}/lib/Alembic.dll"
-                   "\${_IMPORT_PREFIX}/bin/Alembic.dll" RELEASE_CONFIG "${RELEASE_CONFIG}")
-    file(WRITE ${CURRENT_PACKAGES_DIR}/share/${PORT}/AlembicTargets-release.cmake "${RELEASE_CONFIG}")
+
+    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
```
Comment (JackBoosY) [inline ports/alembic/portfile.cmake:49]: What I think is we need to make a patch to fix this thoroughly.

**Comment by dg0yt** [comment]
> it doesn't check that each SHA exists

@BillyONeal Not each SHA. But perhaps it should check that the latest SHA matches the port current build tree?

**Comment by BillyONeal** [comment]
> > it doesn't check that each SHA exists
> 
> @BillyONeal Not each SHA. But perhaps it should check that the latest SHA matches the port current build tree?

Hmmmm that seems reasonable to me but I'd prefer @vicroms to comment

**Comment by JackBoosY** [comment]
Ping @vicroms for review.

**Comment by vicroms** [comment]
> Not each SHA. But perhaps it should check that the latest SHA matches the port current build tree?

There is `--verify-git-tree`

https://github.com/microsoft/vcpkg-tool/blob/f6c8e328d9fd32be36d27e1c3036ef1ec1521148/src/vcpkg/commands.civerifyversions.cpp#L35-L40

Which we may want to enable in CI, but it will majorly slow down the pipeline.

**Review by vicroms — APPROVED** [review]

**Pull request merged**